src/support/hash_deterministic.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef wasm_support_hash_deterministic_h
+#define wasm_support_hash_deterministic_h
```

Comment:
"hash_deterministic" sounds backwards to me. What would you think of using "deterministic_hash" instead?

Reply:
The define here mirrors the file structure of `wasm/support/hash_deterministic.h`, but I guess you are suggesting to rename the file to `deterministic_hash.h` in general? If so, can do, just the files won't order next to each other in an IDE anymore if that's ok.

Reply:
Yeah, sorry, I was mostly thinking about the file name. Unfortunate that they won't be together, but I think `deterministic_hash.h` would read better. Another option would be to put both sets of functionality together in separate sections in `hash.h`. That might make it easier for readers to discover the difference between them, too.